src/lib/mongoBackend/safeBsonGet.cpp

```diff
@@ -104,11 +104,18 @@ int getIntField(const BSONObj& b, const std::string& field)
 *
 * getLongField -
 */
-long getLongField(const BSONObj& b, const std::string& field)
+long long getLongField(const BSONObj& b, const std::string& field)
```

Comment:
Not sure if I like this...

This function shouldn't be so clever. If I'm trying to read a long field that actually is an int, I don't want this function "hides" that: I want to get a Runtime Error that point me where the problem is and fix it. That's the whole idea behind safeBsonGet functions.

It we need "intelligence" to read fields that at the same time may be int or long, then that intelligence has to be put in the caller, not in the get*Field() function.